python_ta/cfg/visitor.py

```diff
@@ -48,9 +54,16 @@ def __getattr__(self, attr: str):
 
     def visit_generic(self, node: nodes.NodeNG) -> None:
         """By default, add the expression to the end of the current block."""
-        self._current_block.add_statement(node)
+        if self._current_block is not None:
+            self._current_block.add_statement(node)
 
     def visit_module(self, module: nodes.Module) -> None:
+        # Modules shouldn't be considered if user specifies functions to render
+        if self._functions_to_render is not None:
+            for child in module.body:
```

Comment:
The overall approach is good but a little more on the permissive side than the efficient side in terms of how it searches for functions.

1. When iterating over the module `body`, add a check so that we only recurse on `FunctionDef`s (with a matching name) and `ClassDef`s (to search for methods).
2. When iterative over a class def `body`, add a check so that we only recurse on `FunctionDef`s with a matching name.

In particular, you don't need to worry about finding `FunctionDef`s that are nested inside other types of syntax, e.g. if statements or even other `FunctionDef`s. It's fine to just look for function defs at the top-level of a module or class definition.